From: Steven Whitehouse <swhiteho@redhat.com>
Date: Wed, 3 Feb 2010 11:34:12 -0500
Subject: [gfs] GFS2: Use GFP_NOFS for alloc structure
Message-id: <1265196852.2370.14.camel@localhost>
Patchwork-id: 23114
O-Subject: [RHEL6] GFS2: Three one liners from upstream (bz #561307) [2/3]
Bugzilla: 561307
RH-Acked-by: Abhijith Das <adas@redhat.com>
RH-Acked-by: Benjamin Marzinski <bmarzins@redhat.com>
RH-Acked-by: Robert S Peterson <rpeterso@redhat.com>

>From ea8d62dadd0217334fb2c5d60e7f89e14076ca10 Mon Sep 17 00:00:00 2001
From: Steven Whitehouse <swhiteho@redhat.com>
Date: Fri, 29 Jan 2010 15:48:57 +0000
Subject: [PATCH] GFS2: Use GFP_NOFS for alloc structure

This is called under a glock, so its a good plan to use GFP_NOFS

Signed-off-by: Steven Whitehouse <swhiteho@redhat.com>
Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/fs/gfs2/rgrp.c b/fs/gfs2/rgrp.c
index 0608f49..d87c0f8 100644
--- a/fs/gfs2/rgrp.c
+++ b/fs/gfs2/rgrp.c
@@ -915,7 +915,7 @@ void gfs2_rgrp_repolish_clones(struct gfs2_rgrpd *rgd)
 struct gfs2_alloc *gfs2_alloc_get(struct gfs2_inode *ip)
 {
 	BUG_ON(ip->i_alloc != NULL);
-	ip->i_alloc = kzalloc(sizeof(struct gfs2_alloc), GFP_KERNEL);
+	ip->i_alloc = kzalloc(sizeof(struct gfs2_alloc), GFP_NOFS);
 	return ip->i_alloc;
 }
 
